### DISCUSSION TITLE
corePlugins safelisting does not work in standalone executable

### DISCUSSION THREAD

**Discussion by esheltonusef** [tailwindlabs/tailwindcss#11953] [Help]
Using tailwindcss-windows-x64.exe 3.3.3, standalone executable, in Visual Studio, .net Core.

When I add [the doc example](https://tailwindcss.com/docs/configuration#core-plugins) to my config file, it breaks the build process. When comment out the the corePlugins section, the build works, and everything else in my config file works accordingly.

```
    corePlugins: [
        'margin',
        'padding',
        'backgroundColor',
    ],   
```

![image](https://github.com/tailwindlabs/tailwindcss/assets/136732994/3c1632e5-f4e1-4821-ae71-6e8d716647a7)

Update:
My expectation when adding core plugins is to have all related classes included in the output so they are readily available during development. This project is not launched, and this implementation is for developers. I am assuming that pseudo/responsive classes will work for these as well. We also have classes in our database that cannot be scanned. I have used the Safelist section extensively but wanted to try a broader option so I don't have to keep using the Safelist until styles are set.

When I build the project in VS I get the below error, which is the Pre-build Event. We delete output.css, then run .exe, set the input and output. (I might not need the Delete at this point but I wouldn't expect this to be the problem).

[MSB3073](https://learn.microsoft.com/en-us/visualstudio/msbuild/errors/msb3073?view=vs-2022&f1url=%3FappId%3DDev16IDEF1%26l%3DEN-US%26k%3Dk(MSBuild.Exec.CommandFailed)%26rd%3Dtrue)	The command "del "wwwroot\css\output.css" & "G:/Class Libraries/Tailwind/tailwindcss-windows-x64.exe" -i ./wwwroot/css/input.css -o ./wwwroot/css/output.css" exited with code 1.

**Comment by wongjn** [comment]
Could you give more details about how "it breaks the build process"? What happens? Is there any error messages you could provide? What do you expect to happen?

**Reply by esheltonusef** [reply]
Thanks for your response. I have updated the original post. The error pertains to the Pre-build Event. My expectations, as stated above, are that the included core plugins would be provided in the output.css and would be readily available in development.

**Reply by wongjn** [reply]
Are there any other messages? The error message does not tell us much. Consider running your commands separately to confirm it is the Tailwind command that is the source of the non-zero exit code. Also, consider running commands with some sort of verbosity or debug flag to perhaps expose more information about the failure.

Also, just so we're on the same page, setting `corePlugins` to an array [disables all other Tailwind core plugins](https://tailwindcss.com/docs/configuration#core-plugins). This could mean that there is some configuration or code that is relying on a core plugin that has been disabled.

Furthermore, core plugins are not available in the output CSS. Core plugins are JavaScript code so one cannot provide them in a CSS file in any meaningful way. Only the CSS they would allow to be produced via Tailwind's CSS rule generation engine would be available in the output CSS and only for the classes that Tailwind has observed to be used in files scanned or classes specified by the `safelist` configuration.

**Reply by esheltonusef** [reply]
That is the only error message.

This is important information. I was conflating the `safelist` configuration and the `corePlugins` configuration, in that I thought they would produce a similar result of forcing additional classes into the output (whether scanned or not), but I was mistaken. I wonder if there is a method to broadly safelist classes rather than listing every variation.

What you said is possible, by using the array and wiping all other `corePlugins` I may have inadvertently caused an unnamed issue leading to the failed build. I will test this further to see if I can isolate the problem.

**Reply by wongjn** [reply]
For broader safelisting of classes, consider using [regular expressions](https://tailwindcss.com/docs/content-configuration#using-regular-expressions).
